Mapify/Utils/Extensions.cs

```diff
@@ -277,6 +277,56 @@ public static void SetBasicMapInfo(this SaveGameData saveGameData, BasicMapInfo
                 saveGameData.SetJObject(SAVE_KEY_NAME, JObject.FromObject(basicMapInfo));
         }
 
+        public static RailTrack GetRailTrack(this RailTrackRegistry registry, string stationID, string yardID, byte trackNumber)
+        {
+            var query = $"[{stationID}]_[{yardID}-{trackNumber:D2}";
+
+            return registry.AllTracks.FirstOrDefault(track => track.name.Contains(query));
+        }
+
+        public static Junction.Branch FindClosestBranch(this Junction junction, Vector3 fromPoint, float maxRange = 5f)
+        {
+            var closestDistance = float.PositiveInfinity;
+
+            RailTrack track = null;
+            var first = false;
+
+            foreach (var foundTrack in Resources.FindObjectsOfTypeAll<RailTrack>())
```

Comment:
We have a lot of calls to `FindObjectsOfType<RailTrack>`. Can we cache these, or retrieve the array from the game's cache (`RailTrackRegistry` or something similar, I forget the exact name)? Calling `FindObjectsOfType` is extremely expensive and allocates a new array every time, so avoiding it wherever possible would be good. If it's too much work or needs hacks, then it's fine as-is.